File: src/main/java/com/pancisin/bookster/models/BookRequest.java
package com.pancisin.bookster.models;

import java.util.Calendar;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Table(name = "book_requests")
public class BookRequest {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;
	
	@JsonIgnore
	@ManyToOne
	private User user;
	
	@ManyToOne
	private Service service;
	
	@Column
	private int units;
	
	@Column
	private Calendar date;

	@Column
	private Boolean approved;
	
	public String getEmail() {
		return getUser().getEmail();
	}
	
	public Long getId() {
		return id;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public Service getService() {
		return service;
	}

	public void setService(Service service) {
		this.service = service;
	}

	public int getUnits() {
		return units;
	}

	public void setUnits(int units) {
		this.units = units;
	}

	public Calendar getDate() {
		return date;
	}

	public void setDate(Calendar date) {
		this.date = date;
	}

	public Boolean getApproved() {
		return approved;
	}

	public void setApproved(Boolean approved) {
		this.approved = approved;
	}
}

File: src/main/kotlin/com/pancisin/bookster/model/BookRequest.kt
package com.pancisin.bookster.model

import java.util.Calendar

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.*

@Entity
@Table(name = "book_requests")
data class BookRequest(

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  val id: Long? = null,

  @JsonIgnore @ManyToOne
  var user: User? = null,

  @ManyToOne
  var service: Service? = null,

  @Column
  var units: Int = 0,

  @Column
  var date: Calendar? = null,

  @Column
  var approved: Boolean? = null,

  @OneToOne
  var submission: FormSubmission? = null

) {
  val email: String?
    get() = user?.email
}
